appsec/src/extension/commands_helpers.c

```diff
@@ -461,17 +461,23 @@ dd_result dd_command_proc_resp_verd_span_data(
     mpack_node_t actions = mpack_node_array_at(root, RESP_INDEX_ACTION_PARAMS);
     dd_result res = _command_process_actions(actions, ctx);
 
-    if (res == dd_should_block || res == dd_should_redirect ||
-        res == dd_should_record) {
+    if (res == dd_should_block || res == dd_should_redirect) {
+        dd_tags_set_sampling_priority();
         dd_trace_emit_asm_event();
         _set_appsec_span_data(
             mpack_node_array_at(root, RESP_INDEX_APPSEC_SPAN_DATA));
     }
 
+    if (res == dd_should_record) {
+        _set_appsec_span_data(
+            mpack_node_array_at(root, RESP_INDEX_APPSEC_SPAN_DATA));
+    }
+
```

Comment:
I'm not sure `dd_should_block` and `dd_should_redirect` should be treated differently, assuming the waf sends `keep` in the correct circumstances